File: StadiumEngine/StadiumEngine.DTO/Accounts/AuthorizeUserDto.cs
using System.Security.Claims;
using Newtonsoft.Json;

namespace StadiumEngine.DTO.Accounts;

/// <summary>
/// ДТО пользователя при авторизации
/// </summary>
public class AuthorizeUserDto
{
    /// <summary>
    /// Полное имя
    /// </summary>
    public string FullName { get; set; }
    
    /// <summary>
    /// Название роли
    /// </summary>
    public string? RoleName { get; set; }
    
    /// <summary>
    /// Флаг суперюзера
    /// </summary>
    public bool IsSuperuser { get; set; }
    
    /// <summary>
    /// Язык
    /// </summary>
    public string Language { get; set; }
    
    [JsonIgnore]
    public List<Claim> Claims { get; set; }
}

File: StadiumEngine/StadiumEngine.Extranet/ClientApp/src/models/dto/accounts/AuthorizeUserDto.ts
import {UserPermissionDto} from "./UserPermissionDto";

export interface AuthorizeUserDto {
    fullName: string,
    roleName: string | null,
    isAdmin: boolean,
    isSuperuser: boolean,
    language: string
    stadiumGroupName: string
    permissions: UserPermissionDto[]
}